packages/tokenbank_v1/contracts/modules/exchanges/LoopringDEXModule.sol

```diff
@@ -0,0 +1,247 @@
+/*
+
+  Copyright 2017 Loopring Project Ltd (Loopring Foundation).
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+pragma solidity ^0.5.11;
+
+import "../../iface/Wallet.sol";
+
+import "../stores/ReimbursementStore.sol";
+
+import "../security/SecurityModule.sol";
+
+import "./IExchangeV3.sol";
+
+
+/// @title LoopringDEXModule
+contract LoopringDEXModule is SecurityModule
+{
+    event AccountUpdated(
+        address indexed exchange,
+        address indexed wallet,
+        bool            creation
+    );
+
+    event Deposit(
+        address indexed exchange,
+        address indexed wallet,
+        address indexed token,
+        uint96          amount
+    );
+
+    event Withdrawal(
+        address indexed exchange,
+        address indexed wallet,
+        address indexed token,
+        uint96          amount
+    );
+
+    ReimbursementStore internal reimbursementStore;
+
+    constructor(
+        SecurityStore      _securityStore,
+        ReimbursementStore _reimbursementStore
+        )
+        public
+        SecurityModule(_securityStore)
+    {
+        reimbursementStore = _reimbursementStore;
+    }
+
+    function staticMethods()
+        public
+        pure
+        returns (bytes4[] memory methods)
+    {
+        methods = new bytes4[](3);
+        methods[0] = this.isExchangeRunning.selector;
+        methods[1] = this.getDEXSlots.selector;
+        methods[2] = this.getDEXAccount.selector;
+    }
+
+    function isExchangeRunning(IExchangeV3 exchange)
+        public
+        view
+        returns (bool)
+    {
+        return (
+            !exchange.isInMaintenance() &&
+            !exchange.isInWithdrawalMode() &&
+            !exchange.isShutdown()
+        );
+    }
+
+    function getDEXSlots(IExchangeV3 exchange)
+        public
+        view
+        returns (
+            uint numAvailableDepositSlots,
+            uint numAvailableWithdrawalSlots
+        )
+    {
+        (,numAvailableDepositSlots, , numAvailableWithdrawalSlots) = exchange.getRequestStats();
+    }
+
+    function getDEXAccount(IExchangeV3 exchange)
+        public
+        view
+        returns (
+            uint24 accountId,
+            uint   pubKeyX,
+            uint   pubKeyY
+        )
+    {
+        return exchange.getAccount(address(this));
```

Comment:
Should use the wallet's address.

Reply:
the method is bound to the wallet, so I think address(this) will be the wallet's address.